components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -286,7 +318,33 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
         // See comment in SHOCPreprocess::operator() about the necessity of tke_copy
         tke(i,k) = tke_copy(i,k);
 
-        qv(i,k) = qw(i,k) - qc(i,k);
+        /*--------------------------------------------------------------------------------
+         *DRY-TO-WET MMRs:
+         *-----------------
+         *Since the host model (or AD) expects wet mixing ratios, we need to convert dry
+         *mixing ratios from SHOC to wet mixing ratios (except TKE).
+
+         *NOTE: Function calculate_wetmmr_from_drymmr takes 2 arguments: ( dry mmr and
+         *"dry" water vapor mixing ratio)
+         *---------------------------------------------------------------------------------
+         */
+
+        //Units of all tracers below will become [kg/kg(wet-air)] for mass and [#/kg(wet-air)]
+        //for number (except TKE)
+
+        const auto qv_dry = qtracers(i,qv_index,k);// store qv "dry" mmr
+
+        //iterate over all tracers and convert dry mmr to wet mmr (except TKE)
+        const int num_qtracer_packs = ekat::npack<Spack>(num_qtracers);
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, num_qtracer_packs), [&] (const Int& q_iter) {
+            const auto is_not_tke_index = q_iter != tke_index;
+            if(is_not_tke_index){PF::calculate_wetmmr_from_drymmr(qtracers(i,q_iter,k),qv_dry);}
+          });
+
+
+        //convert qc from dry to wet mmr[kg/kg(wet-air)]
+        qc(i,k) = PF::calculate_wetmmr_from_drymmr(qc(i,k), qv_dry);
+        qv(i,k) = PF::calculate_wetmmr_from_drymmr(qw(i,k),qv_dry) - qc(i,k);
```

Comment:
`qv` and `qc` are slices of `qtracers`. As such, you are already converting them in the main loop.